### PR TITLE
mergeChanges update

### PR DESCRIPTION
**Pull request by githabas**
It fixes two issues described in https://github.com/vitmalina/w2ui/issues/879

### CODE REVIEW

File: src/w2grid.js [hunk 1]
```diff
@@ -2092,11 +2092,13 @@
                 var record = this.get(changes[c].recid);
                 for (var s in changes[c]) {
                     if (s == 'recid') continue; // do not allow to change recid
-                    var key = "." + s;
-                    if (w2utils.isFloat(s)) key = "['" + s + "']";
-                    if (String(s).indexOf("'") != -1) key = "['" + s.replace(/'/g, "\\'") + "']";
+                    if(typeof changes[c][s] === "object") changes[c][s] = changes[c][s].text;
```
Comment (vitmalina) [inline src/w2grid.js]: need space after if and before (. See https://github.com/vitmalina/w2ui/wiki/Contribution-Requirement

File: src/w2grid.js [hunk 2]
```diff
@@ -2092,11 +2092,13 @@
...
                     try { 
-                        eval('record' + key + ' = changes[c][s]'); 
+			if(~s.indexOf('.')) {
+				eval("record['" + s.replace(/\./g, "']['") + "'] = changes[c][s]");
+			} else {
+				record[s] = changes[c][s];
+			};
```
Comment (vitmalina) [inline src/w2grid.js]: No need for trailing semicolumn

File: src/w2grid.js [hunk 3]
```diff
@@ -2092,11 +2092,13 @@
                 var record = this.get(changes[c].recid);
                 for (var s in changes[c]) {
                     if (s == 'recid') continue; // do not allow to change recid
-                    var key = "." + s;
-                    if (w2utils.isFloat(s)) key = "['" + s + "']";
-                    if (String(s).indexOf("'") != -1) key = "['" + s.replace(/'/g, "\\'") + "']";
+                    if(typeof changes[c][s] === "object") changes[c][s] = changes[c][s].text;
                     try { 
-                        eval('record' + key + ' = changes[c][s]'); 
+			if(~s.indexOf('.')) {
```
Comment (vitmalina) [inline src/w2grid.js]: replace with if (s.indexOf('.') != -1)

**Comment by vitmalina** [comment]
I appreciate your persistence. As this logic becomes more difficult to follow, can you add some unit tests for this method where all use cases are covered. There are already some unit tests in 

qa/tests/w2grid.js

you can follow similar examples as it is for some other methods.

**Comment by githabas** [comment]
Here is jsfiddle with links to updated mergeChanges:
http://jsfiddle.net/5ncv1rb7/9/
The second grid uses nested records.

**Comment by vitmalina** [comment]
Appreciate it. Seems to work nicely. There was some bug in unit test. I have fixed it.

**Pull request merged**